### ISSUE TITLE
`manager.query`'s 2nd param `parameters`'s usage?

### ISSUE THREAD

**Issue by purerosefallen** [typeorm/typeorm#7396]
### What was unclear or otherwise insufficient?

I have no idea how to use `EntityManager.query`'s 2nd parameter `parameters`. It displayed in my IDE, but I have no idea how to use it.

```ts
EntityManager.query(     query: string,     parameters?: any[]): Promise<any>
```

### Recommended Fix

Some database drivers, like MySQL on Node.js, use the symbol `?` to reference the parameters in order. Others, like Postgres, use the symbol `$1 $2` to reference the parameters. In TypeORM's query builder, `:someVars` is used. But it should be an object to use in this way. From what I saw in the IDE, the `parameters` param is an array... I'm very confused.. Looking for help.

### Relevant Database Driver(s)

- [ ] `aurora-data-api`
- [ ] `aurora-data-api-pg`
- [ ] `better-sqlite3`
- [ ] `cockroachdb`
- [ ] `cordova`
- [ ] `expo`
- [ ] `mongodb`
- [ ] `mysql`
- [ ] `nativescript`
- [ ] `oracle`
- [x ] `postgres`
- [ ] `react-native`
- [ ] `sap`
- [ ] `sqlite`
- [ ] `sqlite-abstract`
- [ ] `sqljs`
- [ ] `sqlserver`

### Are you willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time, and I know how to start.
- [x ] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by oae** [comment]
Hi, I also have this problem using oracle. parameters are defined as `any[]` array. But for oracle queries, I can pass an object instead of an array.  I had to add `// @ts-ignore` for now. 

```typescript
const data = await connection.query(
    'select * from table_name where some_column = :column and other_column = :otherColumn',
    // @ts-ignore
    { column: "column_value", otherColumn: "other_column_value" }
);
```

**Comment by bombillazo** [comment]
This is killing the usage of TypeORM with raw queries.

**Comment by Ginden** [comment]
`.query` is a direct interface to underlying database driver. Eg. https://node-postgres.com/features/queries#parameterized-query

Passing object work for Oracle, because underlying driver seemingly supports it.

I'm not sure what is an actual issue here, raw queries are almost always database-specific.

**Comment by bh4r4th** [comment]
I totally agree with @bombillazo. Named parameters syntax to be brought in for Postgres. I fixed my query with similar to @bombillazo approach but looks weird compared to `createQueryBuilder().setParameters()` which supports named parameter injection in query.
